src/cairoUtilFuncGen/storage/dynArrayPop.ts

```diff
@@ -88,31 +89,27 @@ export class DynArrayPopGen extends StringIndexedFuncGen {
 
     const getElemLoc =
       isDynamicArray(elementType) || isMapping(elementType)
-        ? [
-            `let (elem_loc) = ${arrayName}.read(loc, newLen);`,
-            `let (elem_loc) = readId(elem_loc);`,
-          ].join('\n')
-        : `let (elem_loc) = ${arrayName}.read(loc, newLen);`;
+        ? endent`
+            let elem_loc = ${arrayName}::read((loc, newLen));
+            let elem_loc = readId(elem_loc);
+            `
```

Comment:
Even though it's possible, let's do:
```suggestion
            let elem_loc = ${arrayName}::read((loc, newLen));
            elem_loc = readId(elem_loc);
            `
```

Reply:
In order to do that we would need to specify variable as mutable. I think, if it cause so much confusion, that the best idea would be to just rename it

Reply:
Oh you are probably right